src/goto-programs/remove_exceptions.cpp

```diff
@@ -333,8 +339,30 @@ void remove_exceptionst::add_exception_dispatch_sequence(
         // Universal handler. Highest on the stack takes
         // precedence, so overwrite any we've already seen:
         default_target=new_state_pc;
+
+        // Record the position of the default behaviour as any further catches
+        // will not capture the throw
+        default_try=i;
+        default_catch=j;
+        break;
       }
-      else
+    }
+  }
+
+  // add GOTOs implementing the dynamic dispatch of the
+  // exception handlers.
+  // The first loop is in forward order because the insertion reverses the order
+  // we note  that try1{ try2 {} catch2c {} catch2d {}} catch1a() {} catch1b{}
+  // must catch in the following order: 2c 2d 1a 1b hence the numerical index
+  // is reversed whereas the letter ordering remains the same.
+  for(std::size_t i=default_try; i<stack_catch.size(); i++)
+  {
+    std::size_t j;
+    for(j=(i==default_try) ? default_catch : stack_catch[i].size(); j-->0;)
```

Comment:
Isn't all this the same as
`for(std::size_t j=(i==default_try) ? default_catch : stack_catch[i].size(); j>0; --j)` ?